teuthology/orchestra/daemon/systemd.py

```diff
@@ -106,7 +106,7 @@ def pid(self):
         # process regex to match RADOSGW process command string
         # eg. "/usr/bin/radosgw -f --cluster ceph --name <daemon-id=self.id_>"
         if self.type_ == "rgw":
-            proc_regex = "{}.* --name .*{}".format(self.daemon_type, self.id_)
+            proc_regex = '"{}.*--name.*{}"'.format(self.daemon_type, self.id_)
```

Comment:
I wonder if we could use python regex, instead of remote grep, so we just grep 'bin/ceph|bin/radosgw' and parse id on client side. Using python regex will be much easier to test.

Reply:
Since systemd module is specific to a daemon in a node, I think its good to keep same regex. And SystemDstate gives all attributes( including role, id) which makes easier to form regex and get daemon process id and it is very consistent.